PRIME-2869 FNHA Virtual Site Support Part 2 - Individual Enrolment

bergomi02 opened the pull request.

neophyte57 commented:
When I switch College from default "None" to CPSBC, there seems to be excessive calls to web-api and probably not at the correct situation

![image](https://github.com/user-attachments/assets/281bd79f-955a-4425-acbb-37aaaa39118a)

bergomi02 commented:
> When I switch College from default "None" to CPSBC, there seems to be excessive calls to web-api and probably not at the correct situation
> 
> ![image](https://private-user-images.githubusercontent.com/7586665/416726686-281bd79f-955a-4425-acbb-37aaaa39118a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1MDMzNDgsIm5iZiI6MTc0MDUwMzA0OCwicGF0aCI6Ii83NTg2NjY1LzQxNjcyNjY4Ni0yODFiZDc5Zi05NTVhLTQ0MjUtYWNiYi0zN2FhYWEzOTExOGEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjVUMTcwNDA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWJkYzNjODk0NzVkOGU0NDE1YjY1ZTc2OWIwZGY3NzlhY2NmZDc1NGQyNGExZGQ2Njk0YmU2MzA5MWUxMzc5MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.xMrRjG7J-FoXf1Psl101kcen8F61h7zmvkO7hJ1O9QE)

This is fixed.

neophyte57 commented:
If an enrollee has selected a Site that has identified the enrollee as a Remote User, if that Remote User record is later removed from the Site, the connection to the enrollee appears severed 👍  However the `RemoteAccessLocation` record isn't deleted and can show up if the enrollee is able to select the Site again.  Perhaps not in scope of this Jira ticket 🤔

neophyte57 commented:
1. Saved licence class does not match Remote User (e.g. "Full - Specialty").  Change to "Full - Family".  Then Save and Continue
2. Remote Access screen is bypassed when it should not be ✖️
3. Go Back from SD questions screen
4. See "Full - Family" is selected.  Save and Continue.  Remote Access screen is bypassed again.  Go Back. 
5. Now see Remote Access screen
6. Go Back and select "Full - Specialty".  Save and Continue.
7. Error encountered ✖️

![image](https://github.com/user-attachments/assets/f24d3ea5-ce7f-43c2-a520-0b3073f11f0d)

bergomi02 commented:
> 1. Saved licence class does not match Remote User (e.g. "Full - Specialty").  Change to "Full - Family".  Then Save and Continue
> 2. Remote Access screen is bypassed when it should not be ✖️
> 3. Go Back from SD questions screen
> 4. See "Full - Family" is selected.  Save and Continue.  Remote Access screen is bypassed again.  Go Back.
> 5. Now see Remote Access screen
> 6. Go Back and select "Full - Specialty".  Save and Continue.
> 7. Error encountered ✖️
> 
> ![image](https://private-user-images.githubusercontent.com/7586665/416917482-f24d3ea5-ce7f-43c2-a520-0b3073f11f0d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1Mjk3NTYsIm5iZiI6MTc0MDUyOTQ1NiwicGF0aCI6Ii83NTg2NjY1LzQxNjkxNzQ4Mi1mMjRkM2VhNS1jZTdmLTQzYzItYTUyMC0wYjMwNzNmMTFmMGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMDAyNDE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzY2M2VlNjJlYTMxNDQxZjA1MTI5NmFiNjU1OTJkZGIzOTU0ZmQ1MTllY2Q0MTlkZWRhOWZjZjk2YWJlYWFlOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.-fq98yzeykyzX12-GZ7KcrnPBLG7mdHn5MftNWRpptU)



> 1. Saved licence class does not match Remote User (e.g. "Full - Specialty").  Change to "Full - Family".  Then Save and Continue
> 2. Remote Access screen is bypassed when it should not be ✖️
> 3. Go Back from SD questions screen
> 4. See "Full - Family" is selected.  Save and Continue.  Remote Access screen is bypassed again.  Go Back.
> 5. Now see Remote Access screen
> 6. Go Back and select "Full - Specialty".  Save and Continue.
> 7. Error encountered ✖️
> 
> ![image](https://private-user-images.githubusercontent.com/7586665/416917482-f24d3ea5-ce7f-43c2-a520-0b3073f11f0d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1OTIwNjQsIm5iZiI6MTc0MDU5MTc2NCwicGF0aCI6Ii83NTg2NjY1LzQxNjkxNzQ4Mi1mMjRkM2VhNS1jZTdmLTQzYzItYTUyMC0wYjMwNzNmMTFmMGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMTc0MjQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTllNTRiNjFkNGU4MWEyZGZiYzBhODA5ZWYwODQzYWY4OThiYmVjNGM0NjUzZTRiZjg3NTJlMjY0ZmM0MTI2MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.u_iKj-_MNhsqXa9JhtT3TQQ_QV6GiAkCUy586dO1NCU)

The last fix introduced this new issue. It is fixed now.

neophyte57 commented:
1. Saved licence class does not match Remote User (e.g. "Full - Specialty"). Change to "Full - Family". Then Save and Continue
2. Remote Access screen displayed ✔️  Save and Continue without selecting anything 
3. Go Back from SD questions screen
4. Returns to College Licence Information, not Remote Access screen ❓
5. Select "Full - Specialty".  Then Save and Continue
6. Go Back from SD questions screen
7. Now see Remote Access screen.  Error encountered ✖️

![image](https://github.com/user-attachments/assets/b939e73e-0ffb-4288-b356-38e90d38be93)

bergomi02 commented:
> 1. Saved licence class does not match Remote User (e.g. "Full - Specialty"). Change to "Full - Family". Then Save and Continue
> 2. Remote Access screen displayed ✔️  Save and Continue without selecting anything
> 3. Go Back from SD questions screen
> 4. Returns to College Licence Information, not Remote Access screen ❓
> 5. Select "Full - Specialty".  Then Save and Continue
> 6. Go Back from SD questions screen
> 7. Now see Remote Access screen.  Error encountered ✖️
> 
> ![image](https://private-user-images.githubusercontent.com/7586665/417302364-b939e73e-0ffb-4288-b356-38e90d38be93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1OTkxNjEsIm5iZiI6MTc0MDU5ODg2MSwicGF0aCI6Ii83NTg2NjY1LzQxNzMwMjM2NC1iOTM5ZTczZS0wZmZiLTQyODgtYjM1Ni0zOGU5MGQzOGJlOTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMTk0MTAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDdhYjQwM2MzMmM0NDJhNGQwYWU4YWQ0MDlhZTc3YjA5NjBhZWY1ODM4Mzk1OTBlNjIxNjBhOWQ2YmQ3YjJjMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.Mu0fg6w2kR74Z3bufsx1N6ac-FD0mNmPhvPwEYSGMTw)

It should be fixed.

neophyte57 commented:
Doing that behavior again when I switch College from default "None" to CPSBC?

![image](https://github.com/user-attachments/assets/d4415c74-463c-4225-82db-b6d8bd3c2b7e)

bergomi02 commented:
> Doing that behavior again when I switch College from default "None" to CPSBC?
> 
> ![image](https://private-user-images.githubusercontent.com/7586665/417344390-d4415c74-463c-4225-82db-b6d8bd3c2b7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2MDc3NDgsIm5iZiI6MTc0MDYwNzQ0OCwicGF0aCI6Ii83NTg2NjY1LzQxNzM0NDM5MC1kNDQxNWM3NC00NjNjLTQyMjUtODJkYi1iNmQ4YmQzYzJiN2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMjIwNDA4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDBjMDM0N2FjYTQ2NDdiMTgwOWM3Y2UwNzQxYzIxMDU0ZjI1ZGRiZTM1NDZhNjE3YmE2ZTBhNDEwMDk1YzA5YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.RtViUsLpXkaf0bfIeK50hII5oI4PAuWpAzU7MHAm7tw)

Sorry, it should be fixed.

neophyte57 commented:
Something strange still happening, navigating from Care Setting (set to PCHP) to College Licence, getting a torrent of requests to `remote-users`

![image](https://github.com/user-attachments/assets/3b259fed-c9c7-411f-804f-1132073b784c)

bergomi02 commented:
> Something strange still happening, navigating from Care Setting (set to PCHP) to College Licence, getting a torrent of requests to `remote-users`
> 
> ![image](https://private-user-images.githubusercontent.com/7586665/417354157-3b259fed-c9c7-411f-804f-1132073b784c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2MTAyODQsIm5iZiI6MTc0MDYwOTk4NCwicGF0aCI6Ii83NTg2NjY1LzQxNzM1NDE1Ny0zYjI1OWZlZC1jOWM3LTQxMWYtODA0Zi0xMTMyMDczYjc4NGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMjI0NjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODFhM2FkZjJmZGRmY2FiZTE4ZGMxN2M2YmIxNTY1MWE0ZmQ4ZWE0OWI5NDVjMjBlNWZhNmQwZjUzYWExMTg1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ARDvwIEtl1TiFF7UZ3pP4p10pxdvUcLeHGLN09Kd5eg)

Yes, I noticed that. The code to check the logic has been here for long time. I just changed it to call a API. I will see if I can get rid of the excessive API calls.

bergomi02 commented:
> > Something strange still happening, navigating from Care Setting (set to PCHP) to College Licence, getting a torrent of requests to `remote-users`
> > ![image](https://private-user-images.githubusercontent.com/7586665/417354157-3b259fed-c9c7-411f-804f-1132073b784c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2MTAyODQsIm5iZiI6MTc0MDYwOTk4NCwicGF0aCI6Ii83NTg2NjY1LzQxNzM1NDE1Ny0zYjI1OWZlZC1jOWM3LTQxMWYtODA0Zi0xMTMyMDczYjc4NGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMjI0NjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODFhM2FkZjJmZGRmY2FiZTE4ZGMxN2M2YmIxNTY1MWE0ZmQ4ZWE0OWI5NDVjMjBlNWZhNmQwZjUzYWExMTg1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ARDvwIEtl1TiFF7UZ3pP4p10pxdvUcLeHGLN09Kd5eg)
> 
> Yes, I noticed that. The code to check the logic has been here for long time. I just changed it to call a API. I will see if I can get rid of the excessive API calls.

Updated the code to remove the excessive API calls. Also, added the logic to remove the remote access location if the enrollee only have remote access to one site and the site remove user is being removed.

neophyte57 commented:
> > Something strange still happening, navigating from Care Setting (set to PCHP) to College Licence, getting a torrent of requests to `remote-users`
> > ![image](https://private-user-images.githubusercontent.com/7586665/417354157-3b259fed-c9c7-411f-804f-1132073b784c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2MTAyODQsIm5iZiI6MTc0MDYwOTk4NCwicGF0aCI6Ii83NTg2NjY1LzQxNzM1NDE1Ny0zYjI1OWZlZC1jOWM3LTQxMWYtODA0Zi0xMTMyMDczYjc4NGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMjI0NjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODFhM2FkZjJmZGRmY2FiZTE4ZGMxN2M2YmIxNTY1MWE0ZmQ4ZWE0OWI5NDVjMjBlNWZhNmQwZjUzYWExMTg1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ARDvwIEtl1TiFF7UZ3pP4p10pxdvUcLeHGLN09Kd5eg)
> 
> Yes, I noticed that. The code to check the logic has been here for long time. I just changed it to call a API. I will see if I can get rid of the excessive API calls.

Better now!  Thanks Jack

neophyte57 commented:
> > > Something strange still happening, navigating from Care Setting (set to PCHP) to College Licence, getting a torrent of requests to `remote-users`
> > > ![image](https://private-user-images.githubusercontent.com/7586665/417354157-3b259fed-c9c7-411f-804f-1132073b784c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA2MTAyODQsIm5iZiI6MTc0MDYwOTk4NCwicGF0aCI6Ii83NTg2NjY1LzQxNzM1NDE1Ny0zYjI1OWZlZC1jOWM3LTQxMWYtODA0Zi0xMTMyMDczYjc4NGMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMjI0NjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODFhM2FkZjJmZGRmY2FiZTE4ZGMxN2M2YmIxNTY1MWE0ZmQ4ZWE0OWI5NDVjMjBlNWZhNmQwZjUzYWExMTg1ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ARDvwIEtl1TiFF7UZ3pP4p10pxdvUcLeHGLN09Kd5eg)
> > 
> > 
> > Yes, I noticed that. The code to check the logic has been here for long time. I just changed it to call a API. I will see if I can get rid of the excessive API calls.
> 
> Updated the code to remove the excessive API calls. Also, added the logic to remove the remote access location if the enrollee only have remote access to one site and the site remove user is being removed.

Better now! Thanks Jack

neophyte57 reviewed (approved):
Well done!
